src/typ/expr.rs

```diff
@@ -1404,21 +1404,23 @@ impl<'a, 'b, 'c> ExprTyper<'a, 'b, 'c> {
         let constructor = match module {
             // Look in the local scope for a binding with this name
             None => {
-                let constructor = self
-                    .environment
-                    .get_variable(name)
-                    .or_else(|| self.environment.get_module_const(name))
-                    .cloned()
-                    .ok_or_else(|| Error::UnknownVariable {
-                        location: *location,
-                        name: name.to_string(),
-                        variables: self
-                            .environment
-                            .local_values
-                            .keys()
-                            .map(|t| t.to_string())
-                            .collect(),
-                    })?;
+                let constructor = match self.environment.get_variable(name) {
+                    Some(var) => Ok(var.clone()),
+                    None => self
+                        .environment
+                        .get_module_const(name)
+                        .cloned()
+                        .ok_or_else(|| Error::UnknownVariable {
+                            location: *location,
+                            name: name.to_string(),
+                            variables: self
+                                .environment
+                                .local_values
+                                .keys()
+                                .map(|t| t.to_string())
+                                .collect(),
+                        }),
+                }?;
```

Comment:
Had to rearrange this because `get_module_const` now needs an `&mut self` instead of just an `&self` in order to record the const usage